components/Itemlist.js

```diff
@@ -2,27 +2,60 @@ import styled from "styled-components";
 import Link from "next/link";
 import FavoriteIcon from "./FavouriteIcon";
 
-export default function Itemlist({ sneakers, onClick }) {
-  return (
-    <>
-      {sneakers.map((sneaker) => (
-        <StyledItemListContainer key={sneaker.id}>
-          <StyledButton onClick={(event) => onClick(event, sneaker.id)}>
-            <FavoriteIcon isFavorite={sneaker.isFavorite} />
-          </StyledButton>
-
-          <Link href={`/../shoes/${sneaker.id}`}>
-            <img src={sneaker.img} max-width="200px" max-height="200px" />
-            <h2 className="itemName">{sneaker.name}</h2>
-          </Link>
-          <StyledDivContainer>
-            <span className="releaseDate">{sneaker.release}</span>
-            <span className="price">{sneaker.price} €</span>
-          </StyledDivContainer>
-        </StyledItemListContainer>
-      ))}
-    </>
+export default function Itemlist({
+  sneakers,
+  onClick,
+  selectedIcon,
+  setSelectedIcon,
+}) {
+  const filteredSneakers = sneakers.filter(
+    (sneaker) => sneaker.producer.toLowerCase() === selectedIcon
   );
+  if (selectedIcon === "all") {
```

Comment:
wenn ich deine Seite aufrufe, dann erscheint zuerst "no results" - ich muss erst auf "all Sneakers" klicken, damit die angezeigt werden. Bin mir nicht 100% sicher, aber du könntest in deiner _app.js den State schon auf default alle Schuhe setzen mit `const [selectedIcon, setSelectedIcon] = useState("all");` setzen, dann hast du alle, bis du wo anders hin klickst.. das sollte funktionieren...

Reply:
das hatte ich gestern schon umgestellt aber nicht mehr gepusht aber ist jetzt erledigt danke :)